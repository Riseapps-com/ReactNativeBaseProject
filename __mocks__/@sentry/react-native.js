module.exports = {
  ...jest.requireActual('@sentry/react-native'),

  init: jest.fn(),
  captureMessage: jest.fn(),
  captureException: jest.fn(),
};
