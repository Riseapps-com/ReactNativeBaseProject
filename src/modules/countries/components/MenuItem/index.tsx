import React, { useCallback } from 'react';
import { View } from 'react-native';

import { LocalImage, Text } from '~modules/ui';
import { useTheme } from '~theme';

import Card from '../Card';

import themedStyles from './styles';

import type { Images } from '~assets';
import type { StyleProp, ViewStyle } from 'react-native';

export type MenuItemProps = {
  style?: StyleProp<ViewStyle>;
  image: keyof Images;
  title: string;
  index: number;
  onItemPress?: (index: number) => void;
  testID?: string;
};

const MenuItem: React.FC<MenuItemProps> = props => {
  const [styles] = useTheme(themedStyles);
  const { onItemPress } = props;

  const handleItemPress = useCallback(() => onItemPress && onItemPress(props.index), [onItemPress, props.index]);

  return (
    <Card
      testID={props.testID}
      style={[styles.container, props.style]}
      contentStyle={styles.contentContainer}
      onPress={handleItemPress}
      activeOpacity={0.7}
    >
      <View style={styles.imageContainer}>
        <LocalImage style={styles.image} source={props.image} />
      </View>

      <View style={styles.titleContainer}>
        <Text numberOfLines={2} fontStyle="bold" style={styles.title}>
          {props.title}
        </Text>
      </View>
    </Card>
  );
};

export default MenuItem;
