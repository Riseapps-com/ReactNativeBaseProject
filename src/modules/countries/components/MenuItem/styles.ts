import { Platform } from 'react-native';

import { ELEVATION, SHADOW_HEIGHT, SHADOW_OPACITY, SHADOW_RADIUS } from '~modules/ui';
import { sizes, styleSheetFactory } from '~theme';

const MENU_IMAGE_WIDTH = 48;

export default styleSheetFactory(theme => ({
  container: {
    ...Platform.select({
      ios: {
        shadowColor: theme.backdrop,
        shadowOffset: {
          width: 0,
          height: SHADOW_HEIGHT,
        },
        shadowOpacity: SHADOW_OPACITY,
        shadowRadius: SHADOW_RADIUS,
      },
    }),
    marginHorizontal: sizes.SPACING_S,
    marginVertical: sizes.SPACING_S,
    padding: sizes.SPACING_S,
    flex: 1,
  },

  contentContainer: {
    ...Platform.select({ android: { elevation: ELEVATION } }),
    overflow: 'hidden',
    borderRadius: sizes.SPACING_S,
    flexDirection: 'row',
    backgroundColor: theme.primary,
    flex: 1,
  },

  imageContainer: {
    width: '30%',
    alignItems: 'center',
    justifyContent: 'center',
    paddingHorizontal: sizes.SPACING_S,
  },

  image: {
    width: MENU_IMAGE_WIDTH,
    height: MENU_IMAGE_WIDTH,
  },

  titleContainer: {
    flex: 1,
    justifyContent: 'center',
    paddingStart: sizes.SPACING_S,
    paddingEnd: sizes.SPACING_M,
  },

  title: {
    color: theme.secondaryText,
  },
}));
