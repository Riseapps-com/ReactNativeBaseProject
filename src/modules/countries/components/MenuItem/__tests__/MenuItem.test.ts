import { fireEvent } from '@testing-library/react-native';

import { renderComponent } from '~modules/tests';

import MenuItem from '../index';

import type { MenuItemProps } from '../index';

const renderMenuItem = (props: MenuItemProps) => renderComponent(MenuItem, props);

describe('countries', () => {
  describe('<MenuItem />', () => {
    it('renders <MenuItem />', () => {
      const title = 'Title';
      const menuItem = renderMenuItem({
        image: 'africa',
        title,
        index: 0,
      });

      jest.runAllTimers();

      expect(menuItem.getByText(title)).toBeTruthy();
      expect(menuItem).toMatchSnapshot();
    });

    it('calls on item press', () => {
      const testID = 'testID';
      const onItemPress = jest.fn();
      const index = 5;
      const menuItem = renderMenuItem({
        image: 'africa',
        title: 'Title',
        index,
        onItemPress,
        testID,
      });

      fireEvent.press(menuItem.getByTestId(testID));
      jest.runAllTimers();

      expect(onItemPress).toBeCalledTimes(1);
      expect(onItemPress).toBeCalledWith(index);
    });
  });
});
