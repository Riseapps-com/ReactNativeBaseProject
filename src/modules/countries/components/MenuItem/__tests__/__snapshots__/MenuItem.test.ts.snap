// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`countries <MenuItem /> renders <MenuItem /> 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flex": 1,
      "marginHorizontal": 8,
      "marginVertical": 8,
      "opacity": 1,
      "padding": 8,
      "shadowColor": "#000000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.25,
      "shadowRadius": 3.84,
    }
  }
>
  <View
    accessibilityLabel="Content container"
    style={
      Array [
        Object {
          "borderRadius": 8,
          "overflow": "hidden",
        },
        Object {
          "backgroundColor": "#0036A7",
          "flex": 1,
          "flexDirection": "row",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "paddingHorizontal": 8,
          "width": "30%",
        }
      }
    >
      <Image
        accessibilityLabel="[object Object]"
        source={
          Object {
            "testUri": "../../../assets/images/selectRegionMenu/africa.png",
          }
        }
        style={
          Array [
            false,
            false,
            Object {
              "height": 48,
              "width": 48,
            },
          ]
        }
      />
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "justifyContent": "center",
          "paddingEnd": 16,
          "paddingStart": 8,
        }
      }
    >
      <Text
        numberOfLines={2}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            Object {
              "textAlign": "left",
            },
            Array [
              Object {
                "color": "#4A4E4D",
              },
              Object {
                "fontFamily": "SF Pro Text Bold",
              },
              Object {
                "fontSize": 14,
              },
              Object {
                "color": "#FFFFFF",
              },
            ],
          ]
        }
      >
        Title
      </Text>
    </View>
  </View>
</View>
`;
